src/CountryProvider.php

```diff
@@ -92,7 +91,11 @@ private function findCountryCode(Request $request): string
         }
 
         try {
-            return $this->reader->country($request->getClientIp())->country->isoCode ?: $this->fallbackCountry;
+            if (null === $this->reader && !empty($_SERVER['GEOIP2_DATABASE'])) {
+                $this->reader = new Reader($_SERVER['GEOIP2_DATABASE']);
```

Comment:
I don't think this would work because the next line would expect a callback?

Reply:
True, I did not catch this as I didn't test the env var approach.